presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -265,12 +338,27 @@ public Page getOutput()
         return types.build();
     }
 
-    private static class GroupByHashAggregationBuilder
+    private static class GroupByHashAggregationBuilder implements Closeable
     {
-        private final GroupByHash groupByHash;
-        private final List<Aggregator> aggregators;
+        private static final Logger log = Logger.get(GroupByHashAggregationBuilder.class);
+        private static final int MAX_GROUPS_COUNT_DURING_MERGE = 1_000_000;
```

Comment:
shouldn't that be configurable as other variables? why is that variable anyway? what happens when we run out of memory during merge phase?